pegasus/helpers/hourofcode_helpers.rb

```diff
@@ -105,11 +106,13 @@ def hoc_detect_country
   country_code
 end
 
+# Used to set @language instance variable in short format (ex. 'en')
 def hoc_detect_language
   language = request.env['rack.locale']
-  return language if HOC_I18N.keys.include?(language)
+  return nil unless language
   language = language[0..1]
-  return language if HOC_I18N.keys.include?(language)
+
+  return language if I18n.backend.translations[language.to_sym]
```

Comment:
Let's not change the logic of this function in this PR since it isn't related to the string-tracking purpose. 
The old logic accepts languages with length greater than 2 while the new logic does not.